lib/rspec/matchers/dsl.rb

```diff
@@ -80,11 +77,20 @@ def match(&match_block)
         # @yield [Object] actual the actual value (i.e. the value wrapped by `expect`)
         def match_when_negated(&match_block)
           define_user_override(:does_not_match?, match_block) do |actual|
-            @actual = actual
-            super(*actual_arg_for(match_block))
+            begin
+              @actual = actual
+              RSpec::Support.with_failure_notifier(RAISE_METHOD) do
+                super(*actual_arg_for(match_block))
+              end
+            rescue RSpec::Expectations::ExpectationNotMetError
+              false
```

Comment:
> No it should return true as this is the inverse case.

That doesn't sound right to me.  `does_not_match?` is invoked when `expect(x).not_to matcher` is used.  If it returns `true` then that means the expectation will pass.  If you use an expectation within `match_when_negated`, then `does_not_match?` should return `true` if the expectation passes and `false` if the expectation fails -- so within the `rescue` it should return false, I think.